nidx/src/indexer.rs

```diff
@@ -328,13 +328,20 @@ pub async fn index_resource(
     // and the index message can be safely reprocessed
     let mut tx = meta.transaction().await?;
     let mut indexes = Vec::new();
-    for (segment, size, deletions) in results?.into_iter().flatten() {
-        // Mark the segments as visible and write the deletions at the same time
-        segment.mark_ready(&mut *tx, size as i64).await?;
+    for (index_id, segment_result, deletions) in results?.into_iter() {
+        let mut update_index = false;
+        if segment_result.is_some() {
```

Comment:
`if let Some((segment,size)) = segment_result` and then you can remove the next line.